spec.html

```diff
@@ -3784,7 +3780,30 @@ <h1>Runtime Semantics: MV</h1>
               The MV of <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits ExponentPart</emu-grammar> is the MV of |DecimalDigits| times 10<sup>_e_</sup>, where _e_ is the MV of |ExponentPart|.
             </li>
           </ul>
-          <p>Once the exact MV for a String numeric literal has been determined, it is then rounded to a value of the Number type. If the MV is 0, then the rounded value is *+0* unless the first non white space code point in the String numeric literal is `"-"`, in which case the rounded value is *-0*. Otherwise, the rounded value must be the Number value for the MV (in the sense defined in <emu-xref href="#sec-ecmascript-language-types-number-type"></emu-xref>), unless the literal includes a |StrUnsignedDecimalLiteral| and the literal has more than 20 significant digits, in which case the Number value may be either the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit or the Number value for the MV of a literal produced by replacing each significant digit after the 20th with a 0 digit and then incrementing the literal at the 20th digit position. A digit is significant if it is not part of an |ExponentPart| and</p>
+          <p>For the MV of any production not shown above, see <emu-xref href="#sec-static-semantics-mv"></emu-xref>.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-stringtonumber" aoid="StringToNumber">
+          <h1>StringToNumber ( _str_ )</h1>
+          <emu-alg>
+            1. Assert: Type(_str_) is String.
+            1. Let _text_ be the sequence of Unicode code points that results from interpreting _str_ as UTF-16 encoded Unicode text as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>.
+            1. Let _literal_ be the result of parsing _text_ using the goal symbol |StringNumericLiteral|. If _text_ does not conform to the grammar, or if any elements of _text_ were not matched by the parse, return *NaN*.
+            1. NOTE: The terminal symbols of the |StringNumericLiteral| grammar are all code points in the Unicode Basic Multilingual Plane (BMP). Therefore, this operation will return *NaN* if _str_ contains any <emu-xref href="#leading-surrogate"></emu-xref> or <emu-xref href="#trailing-surrogate"></emu-xref> code units, whether paired or unpaired.
```

Comment:
I think this would be phrased better as an assertion, e.g.
```suggestion
            1. Assert: _str_ does not contain any <emu-xref href="#leading-surrogate"></emu-xref> or <emu-xref href="#trailing-surrogate"></emu-xref> code units.
```

Reply:
I'm doubtful that that's better, but I'll change it if there's consensus that it is.

Reply:
It does read much simpler to me.

Reply:
Well, sure, the suggested assertion is *simpler*, because it doesn't contain the reasoning that's in the Note. Even simpler would be to leave out the Note/Assertion entirely. It's only there because the status quo has the Note, and I didn't want to leave anything out. I'm not sure it actually helps the reader understand the algorithm.

And actually, the day may come when it's not true, if Unicode ever defines an astral code point with General_Category=Zs, because then that will be a valid WhiteSpace. (I don't see a guarantee that they won't, though I haven't looked very hard.) And if that ever happens, the algorithm will still run fine, except that the Note/Assertion will be wrong.

So I'm coming around to thinking that the spec is better off without that Note/Assertion.

Reply:
I've added a commit to remove the Note.